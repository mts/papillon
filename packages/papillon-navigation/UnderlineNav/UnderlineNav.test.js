import React from 'react';
import {
  underlineNav,
  underlineNavActions,
  underlineNavRight,
  underlineNavRightWithActions,
  underlineNavWithCounter
} from './UnderlineNav.render';

describe('<Navigation />', () => {
  let wrapper;

  describe('Instance', () => {
    test('must be an instance of UnderlineNav', () => {
      wrapper = renderMount(underlineNav);
      expect(wrapper.find('UnderlineNav')).toHaveLength(1);
    });
  });

  describe('Snaphot', () => {
    test('must match underlineNav', () => {
      expect(global.renderToJSON(underlineNav)).toMatchSnapshot();
    });

    test('must match underlineNavActions', () => {
      expect(global.renderToJSON(underlineNavActions)).toMatchSnapshot();
    });

    test('must match underlineNavRight', () => {
      expect(global.renderToJSON(underlineNavRight)).toMatchSnapshot();
    });

    test('must match underlineNavRightWithActions', () => {
      expect(global.renderToJSON(underlineNavRightWithActions)).toMatchSnapshot();
    });

    test('must match underlineNavWithCounter', () => {
      expect(global.renderToJSON(underlineNavWithCounter)).toMatchSnapshot();
    });
  });
});
