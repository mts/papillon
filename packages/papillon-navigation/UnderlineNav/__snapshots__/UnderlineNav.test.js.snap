// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Snaphot must match underlineNav 1`] = `
<nav
  aria-label=""
  className="UnderlineNav"
>
  <div
    className="UnderlineNav-body"
  >
    <a
      className="UnderlineNav-item selected"
      href="#url"
      role="tab"
      title="Item 1"
    >
      Item 1
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 2"
    >
      Item 2
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 3"
    >
      Item 3
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 4"
    >
      Item 4
    </a>
  </div>
</nav>
`;

exports[`<Navigation /> Snaphot must match underlineNavActions 1`] = `
<nav
  aria-label="Foo bar"
  className="UnderlineNav"
>
  <div
    className="UnderlineNav-body"
  >
    <a
      className="UnderlineNav-item selected"
      href="#url"
      role="tab"
      title="Item 1"
    >
      Item 1
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 2"
    >
      Item 2
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 3"
    >
      Item 3
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 4"
    >
      Item 4
    </a>
  </div>
  <div
    className="UnderlineNav-actions"
  >
    <a
      className="btn"
      href="https://github.com/mts/papillon/tree/master/packages/papillon-navigation/UnderlineNav"
    >
      Button
    </a>
  </div>
</nav>
`;

exports[`<Navigation /> Snaphot must match underlineNavRight 1`] = `
<nav
  aria-label=""
  className="UnderlineNav UnderlineNav--right"
>
  <div
    className="UnderlineNav-body"
  >
    <a
      className="UnderlineNav-item selected"
      href="#url"
      role="tab"
      title="Item 1"
    >
      Item 1
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 2"
    >
      Item 2
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 3"
    >
      Item 3
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 4"
    >
      Item 4
    </a>
  </div>
</nav>
`;

exports[`<Navigation /> Snaphot must match underlineNavRightWithActions 1`] = `
<nav
  aria-label="Foo bar"
  className="UnderlineNav UnderlineNav--right"
>
  <div
    className="UnderlineNav-actions"
  >
    <a
      className="btn"
      href="https://github.com/mts/papillon/tree/master/packages/papillon-navigation/UnderlineNav"
    >
      Button
    </a>
  </div>
  <div
    className="UnderlineNav-body"
  >
    <a
      className="UnderlineNav-item selected"
      href="#url"
      role="tab"
      title="Item 1"
    >
      Item 1
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 2"
    >
      Item 2
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 3"
    >
      Item 3
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 4"
    >
      Item 4
    </a>
  </div>
</nav>
`;

exports[`<Navigation /> Snaphot must match underlineNavWithCounter 1`] = `
<nav
  aria-label="Foo bar"
  className="UnderlineNav"
>
  <div
    className="UnderlineNav-body"
  >
    <a
      className="UnderlineNav-item selected"
      href="#url"
      role="tab"
      title="Item 1"
    >
      <svg
        aria-hidden="true"
        className="UnderlineNav-octicon mr-1"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          d="M4.48 7.27c.26.26 1.28 1.33 1.28 1.33l.56-.58-.88-.91 1.69-1.8s-.76-.74-.43-.45c.32-1.19.03-2.51-.87-3.44C4.93.5 3.66.2 2.52.51l1.93 2-.51 1.96-1.89.52-1.93-2C-.19 4.17.1 5.48 1 6.4c.94.98 2.29 1.26 3.48.87zm6.44 1.94l-2.33 2.3 3.84 3.98c.31.33.73.49 1.14.49.41 0 .82-.16 1.14-.49.63-.65.63-1.7 0-2.35l-3.79-3.93zM16 2.53L13.55 0 6.33 7.46l.88.91-4.31 4.46-.99.53-1.39 2.27.35.37 2.2-1.44.51-1.02L7.9 9.08l.88.91L16 2.53z"
          fillRule="evenodd"
        />
      </svg>
      Item 1
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 2"
    >
      <svg
        aria-hidden="true"
        className="UnderlineNav-octicon mr-1"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          d="M4.48 7.27c.26.26 1.28 1.33 1.28 1.33l.56-.58-.88-.91 1.69-1.8s-.76-.74-.43-.45c.32-1.19.03-2.51-.87-3.44C4.93.5 3.66.2 2.52.51l1.93 2-.51 1.96-1.89.52-1.93-2C-.19 4.17.1 5.48 1 6.4c.94.98 2.29 1.26 3.48.87zm6.44 1.94l-2.33 2.3 3.84 3.98c.31.33.73.49 1.14.49.41 0 .82-.16 1.14-.49.63-.65.63-1.7 0-2.35l-3.79-3.93zM16 2.53L13.55 0 6.33 7.46l.88.91-4.31 4.46-.99.53-1.39 2.27.35.37 2.2-1.44.51-1.02L7.9 9.08l.88.91L16 2.53z"
          fillRule="evenodd"
        />
      </svg>
      Item 2
      <span
        className="ml-1"
      >
        <span
          className="Counter"
        >
          10
        </span>
      </span>
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 3"
    >
      <svg
        aria-hidden="true"
        className="UnderlineNav-octicon mr-1"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          d="M4.48 7.27c.26.26 1.28 1.33 1.28 1.33l.56-.58-.88-.91 1.69-1.8s-.76-.74-.43-.45c.32-1.19.03-2.51-.87-3.44C4.93.5 3.66.2 2.52.51l1.93 2-.51 1.96-1.89.52-1.93-2C-.19 4.17.1 5.48 1 6.4c.94.98 2.29 1.26 3.48.87zm6.44 1.94l-2.33 2.3 3.84 3.98c.31.33.73.49 1.14.49.41 0 .82-.16 1.14-.49.63-.65.63-1.7 0-2.35l-3.79-3.93zM16 2.53L13.55 0 6.33 7.46l.88.91-4.31 4.46-.99.53-1.39 2.27.35.37 2.2-1.44.51-1.02L7.9 9.08l.88.91L16 2.53z"
          fillRule="evenodd"
        />
      </svg>
      Item 3
    </a>
    <a
      className="UnderlineNav-item"
      href="#url"
      role="tab"
      title="Item 4"
    >
      <svg
        aria-hidden="true"
        className="UnderlineNav-octicon mr-1"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          d="M4.48 7.27c.26.26 1.28 1.33 1.28 1.33l.56-.58-.88-.91 1.69-1.8s-.76-.74-.43-.45c.32-1.19.03-2.51-.87-3.44C4.93.5 3.66.2 2.52.51l1.93 2-.51 1.96-1.89.52-1.93-2C-.19 4.17.1 5.48 1 6.4c.94.98 2.29 1.26 3.48.87zm6.44 1.94l-2.33 2.3 3.84 3.98c.31.33.73.49 1.14.49.41 0 .82-.16 1.14-.49.63-.65.63-1.7 0-2.35l-3.79-3.93zM16 2.53L13.55 0 6.33 7.46l.88.91-4.31 4.46-.99.53-1.39 2.27.35.37 2.2-1.44.51-1.02L7.9 9.08l.88.91L16 2.53z"
          fillRule="evenodd"
        />
      </svg>
      Item 4
    </a>
  </div>
</nav>
`;
